test/rules/strict-boolean-expressions/test.ts.lint

```diff
@@ -0,0 +1,178 @@
+class C { }
+enum MyEnum {
+	A, B, C
+}
+let anyType: {};
+let boolType: boolean;
+let bwrapType: Boolean;
+let numType = 9;
+let strType = "string";
+let objType = new Object();
+let classType = new C();
+let enumType = MyEnum.A;
+let boolFn = () => { return true; };
+let strFn = () => { return strType; };
+let numFn = () => { return numType; };
+let boolExpr = (strType !== undefined);
+
+/*** Binary Expressions ***/
+/*** Invalid Boolean Expressions ***/
+classType && boolType;
+~~~~~~~~~              [Operands for the && or || operator should be of type boolean]
+anyType && boolType;
+~~~~~~~              [Operands for the && or || operator should be of type boolean]
+numType && boolType;
+~~~~~~~              [Operands for the && or || operator should be of type boolean]
+boolType && strType;
+            ~~~~~~~  [Operands for the && or || operator should be of type boolean]
+boolType && objType && enumType;
+~~~~~~~~~~~~~~~~~~~              [Operands for the && or || operator should be of type boolean]
+                       ~~~~~~~~  [Operands for the && or || operator should be of type boolean]
+bwrapType && boolType;
+~~~~~~~~~              [Operands for the && or || operator should be of type boolean]
+boolType || classType;
+            ~~~~~~~~~  [Operands for the && or || operator should be of type boolean]
+boolType || anyType;
+            ~~~~~~~  [Operands for the && or || operator should be of type boolean]
+boolType || numType;
+            ~~~~~~~  [Operands for the && or || operator should be of type boolean]
+strType || boolType;
+~~~~~~~              [Operands for the && or || operator should be of type boolean]
+bwrapType || boolType;
+~~~~~~~~~              [Operands for the && or || operator should be of type boolean]
+objType || boolType || enumType;
+~~~~~~~~~~~~~~~~~~~              [Operands for the && or || operator should be of type boolean]
```

Comment:
I don't think it makes sense to underline `objType || boolType`, `objType` should only be underlined. If we can avoid flagging this, we can go back to the way it was before using `errorFlagged` and call `super` since that address the issue of overlapping errors